### DISCUSSION TITLE
Epic Stack + OpenID Connect example (SSO)

### DISCUSSION THREAD

**Discussion by kentcdodds** [epicweb-dev/epic-stack#388] [Announcements]
UPDATE: The Epic Stack's GitHub auth implementation has been altered slightly to make it much easier to add/remove whatever providers you desire. Check the commit: https://github.com/epicweb-dev/epic-stack/commit/1d00e19bea55d01bdb28305a9d56e4eb51ab025a

I'm excited to share an example of using OpenID Connect with Google on the Epic Stack! 🎉 This should be helpful for anyone wanting to support Single-Sign On!

The cool thing about this example is it works with *any* `remix-auth` strategy as well, so if you're trying to integrate with an enterprise using SAML for their SSO strategy, you could use [remix-auth-sso](https://github.com/boxyhq/remix-auth-sso).

I recorded a video demonstrating this example that you can watch here:

[![A frame from a screen cast video of Kent demonstrating the connections page from this demo](https://github-production-user-asset-6210df.s3.amazonaws.com/1500684/260885255-938f0150-51a6-47ae-a9ae-daa09c0c6b9d.png)](https://www.epicweb.dev/tips/add-open-id-connect-auth-to-the-epic-stack)

Find the example here: [github.com/kentcdodds/epic-oidc](https://github.com/kentcdodds/epic-oidc).

Cheers!

P.S. It's very possible I'll adapt some of this into the Epic Stack to make the providers thing generic so it's much easier to plug in your own providers as you wish. This could eventually be made into a library as well possibly...

**Comment by victorferreira** [comment]
Does it make sense to have a examples repo, similar to the [Remix](https://github.com/remix-run/examples) one?

**Reply by kentcdodds** [reply]
I've considered this, but the trouble is I don't have time to maintain them (especially those contributed by others), so for now we're not going to do that. Thanks though!

**Reply by cr101** [reply]
Great idea. An example of using OpenID Connect with LinkedIn would be awesome

**Reply by kentcdodds** [reply]
I imagine it'll look just like this: https://github.com/kentcdodds/epic-oidc/blob/0f23b1cc16c57d9d20a8d17411a61203e4eea94e/app/utils/providers/google.server.ts
